optional_plugins/result_upload/avocado_result_upload/__init__.py

```diff
@@ -0,0 +1,109 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Virtuozzo Inc. 2017
+# Authors: Dmitry Monakhov <dmonakhov@openvz.org>
+
+"""
+Avocado Plugin to propagate Job results to remote host
+"""
+
+import os
+import sys
+import time
+
+import resultsdb_api
+from six import iteritems
+
+from avocado.core.plugin_interfaces import CLI, Result
+
+from avocado.core.settings import settings
+from avocado.utils import process
+
+
+class ResultUpload(Result):
+
+    """
+    ResultsUpload output class
+    """
+
+    name = 'result_upload'
+    description = 'ResultUpload result support'
+
+    def render(self, result, job):
+        """
+        Upload result, which corresponds to one test from
+        the Avocado Job
+
+        if job.status == "RUNNING":
+            return  # Don't create results on unfinished jobs
+
+        """
+        self.upload_url = None
+        if getattr(job.args, 'result_upload_url', None) is not None:
+            self.upload_url = job.args.result_upload_url
+
+        self.upload_cmd = None
+        if getattr(job.args, 'result_upload_cmd', None) is not None:
+            self.upload_cmd = job.args.result_upload_cmd
+
+        if self.upload_url is None:
+            return
+        if self.upload_url is None:
+            return
+        ret = process.run("%s %s %s" % (self.upload_cmd, job.logdir,
+                                        self.upload_url))
+        if ret.exit_status:
+            job.log.error("ResultUploader failed msg=%s" % result.stderr)
```

Comment:
Good. I'd add a log message in case of success as well, so users not have to blindly trust it.

But there's a biggest concern here: Avocado is not interactive, so we have to handle the requests of interaction instead of just interrupting the execution waiting for user input. That would require us to, instead of allowing any command to upload the logs, to have a standard way to do that. As an example, see the remote plugin:
```
$ avocado run passtest.py --remote-hostname=127.0.0.1
JOB ID     : 6e78650acb9992c897a683c74d65c209d40fe4a9
JOB LOG    : /home/apahim/avocado/job-results/job-2017-11-23T14.16-6e78650/job.log
LOGIN      : apahim@127.0.0.1:22 (TIMEOUT: 60 seconds)

Avocado job failed: JobError: Needed to prompt for a connection or sudo password (host: 127.0.0.1), but abort-on-prompts was set to True
$
```

While with this plugin:
```
$ avocado run passtest.py --result-upload-url=127.0.0.1:/tmp/
JOB ID     : e124f0458372a2b711dc8ed513398d9063cd45e5
JOB LOG    : /home/apahim/avocado/job-results/job-2017-11-23T14.17-e124f04/job.log
 (1/1) passtest.py:PassTest.test: PASS (0.04 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 0.96 s
apahim@127.0.0.1's password: 
```
That of course would break the simplicity you're proposing here, but that is the way to go. The Job shall never wait for user interaction in order to complete the execution.